src/main/java/org/springframework/data/elasticsearch/core/query/BulkOptions.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.data.elasticsearch.core.query;
+
+import java.util.List;
+
+import org.elasticsearch.action.support.ActiveShardCount;
+import org.elasticsearch.action.support.WriteRequest;
+import org.elasticsearch.common.unit.TimeValue;
+
+/**
+ * Options that may be passed to an
+ * {@link org.springframework.data.elasticsearch.core.ElasticsearchOperations#bulkIndex(List, BulkOptions)} or
+ * {@link org.springframework.data.elasticsearch.core.ElasticsearchOperations#bulkUpdate(List, BulkOptions)} call.
+ *
+ * @author Peter-Josef Meisch
+ */
+public class BulkOptions {
```

Comment:
With having a Builder in place, what's the reason for non-`final` fields?

Reply:
+1, pass the builder to a private ctor of `BulkOptions` and set the fields there.

Reply:
It should be rather the other way round, to have a private constructor accepting all `BulkOptions` fields to match Spring Data's style.